src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -462,18 +504,14 @@ uct_cuda_ipc_mem_reg(uct_md_h md, void *address, size_t length,
                      const uct_md_mem_reg_params_t *params, uct_mem_h *memh_p)
 {
     uct_cuda_ipc_memh_t *memh;
-    CUdevice cu_device;
-
-    UCT_CUDA_IPC_GET_DEVICE(cu_device);
 
     memh = ucs_malloc(sizeof(*memh), "uct_cuda_ipc_memh_t");
     if (NULL == memh) {
         ucs_error("failed to allocate memory for uct_cuda_ipc_memh_t");
         return UCS_ERR_NO_MEMORY;
     }
 
-    memh->dev_num = (int) cu_device;
-    memh->pid     = getpid();
+    memh->pid = getpid();
```

Comment:
maybe init dev_num to invalid value?
also pls add comment it is initialized during add_reg